bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -720,8 +720,7 @@ public void asyncAddEntry(final byte[] data, final AddCallback cb,
      * @param ctx
      *            some control object
      */
-    public void asyncAddEntry(final long entryId, final byte[] data, final AddCallback cb, final Object ctx)
-            throws BKException {
+    public void asyncAddEntry(final long entryId, final byte[] data, final AddCallback cb, final Object ctx) {
```

Comment:
since we are touching this function now, I'd like to also raise one discussing about the semantic here. Currently it is totally the application's responsibility on managing the semantic on using this API. it would be good to add some constraints on this API:

- if entryId is negative  (the fix @eolivelli tried to fix)
- if entryId is smaller than LAC, that means this entry is a duplicated entry

in both cases, we should fail the addEntry with exceptions.